### PR TITLE
Feature/tvist1 237 document logging

### PR DESCRIPTION
**Pull request by jekuaitk**
- Added documentation on logging
- Modified CHANGELOG
- Added CaseListener
- Added AbstractEntityListener
- Added LoggableEntityInterface
- Added ItkDevLoggingException
- Added Board- and MunicipalityListener
- Added test for Listener logic
- Added AbstractRelatedToCaseListener

### CODE REVIEW

File: docs/LOGGING.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+For storing logs in our database we make a LogEntry entity
+with the necessary fields. That is
+
+* Case ID
+* Entity
+* Entity ID
+* Action
+* User
+* Data
+* Timestamp
```
Comment (lats1) [inline docs/LOGGING.md:33]: Should this be created_at instead?

File: src/Entity/LogEntry.php [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+    /**
+     * @ORM\Column(type="json")
+     */
+    private $data;
+
+    /**
+     * @Gedmo\Timestampable(on="create")
+     * @ORM\Column(type="datetime")
+     */
+    private $timeStamp;
```
Comment (lats1) [inline src/Entity/LogEntry.php:53]: Should this be created_at instead?

File: tests/Logging/AbstractListenerTest.php [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+        // Assert all properties are as expected:
+        $this->assertSame($mockCaseIDString, $logEntry->getCaseID());
+        $this->assertSame(get_class($mockMunicipality), $logEntry->getEntityType());
+        $this->assertSame($mockMunicipalityIDString, $logEntry->getEntityID());
+        $this->assertSame($testAction, $logEntry->getAction());
+        $this->assertSame($mockUsername, $logEntry->getUser());
+        $this->assertSame($expectedDataArray, $logEntry->getData());
+    }
+
+    public function testNonSimpleCreateLogEntry()
```
Comment (lats1) [inline tests/Logging/AbstractListenerTest.php:178]: This name of the method should maybe be more descriptive?

File: tests/Logging/AbstractListenerTest.php [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+        $this->assertSame(get_class($mockCase), $logEntry->getEntityType());
+        $this->assertSame($mockCaseID, $logEntry->getEntityID());
+        $this->assertSame($testAction, $logEntry->getAction());
+        $this->assertSame($mockUsername, $logEntry->getUser());
+        $this->assertSame($expectedDataArray, $logEntry->getData());
+    }
+
+    public function testHandleLoggablePropertiesException()
+    {
+        $this->expectException(ItkDevLoggingException::class);
```
Comment (lats1) [inline tests/Logging/AbstractListenerTest.php:272]: Consider using a more specific exception here.

**Review by lats1 — CHANGES_REQUESTED** [review]

File: docs/LOGGING.md (hunk 1 above)
Comment (jekuaitk) [reply docs/LOGGING.md:33]: Renamed property to createdAt

File: src/Entity/LogEntry.php (hunk 2 above)
Comment (jekuaitk) [reply src/Entity/LogEntry.php:53]: Renamed property to createdAt

File: tests/Logging/AbstractListenerTest.php (hunk 4 above)
Comment (jekuaitk) [reply tests/Logging/AbstractListenerTest.php:272]: Introduced new `ItkDevGetFunctionNotFoundException` exception to be thrown in the `handleLoggableEntities()` function

File: tests/Logging/AbstractListenerTest.php (hunk 3 above)
Comment (jekuaitk) [reply tests/Logging/AbstractListenerTest.php]: Refactored tests to be more readable and test methods to be more telling

**Review by lats1 — APPROVED** [review]

**Pull request merged**